### PR TITLE
burn chart skipping date

### PR DESCRIPTION
**Pull request by TheMaja**
- initial commit
- currentUserBurndownStats filter userId using auth
- date duplicates removed from burn chart
- timestamp used to filter date duplicates
- duplicate dates removed from the burn down chat
- util file with gap function created
- dates added on the x-axis
- dates sorted on chart and demo data is displayed
- array split into sub-arrays
- working uncleaned version
- date duplicates removed and dates filled between gap dates

Related issues: [please specify]

## Description:

What are you up to? Fill us in :)

## Screenshots/videos
**Chart with activity Dots**
chart with dots
**Current version with activity Dots removed looks clean**
dots removed

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have merged the develop branch into my branch and fixed any merge conflicts
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/components/regions/UserActions/UserBurndownStats.jsx [hunk 1]
```diff
@@ -23,21 +24,35 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 export default ({ burnDownSnapshots }) => {
+  burnDownSnapshots = burnDownSnapshots.reduce((unique, o) => {
```
Comment (Sbonelo01) [inline frontend/src/components/regions/UserActions/UserBurndownStats.jsx]: For readability, Can you please better name the arguments taken by the reduce method? The method takes the current value and adds it to the accumulating value: `reduce(acc, curr)`

`unique` is a good name, but `o` isn't. Please just fix that.

File: frontend/src/components/regions/UserActions/utils.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  const allDatesArr = [];
+
+  while (firstDate <= lastDate) {
+    allDatesArr.push(new Date(firstDate).toISOString().split("T")[0]);
+    firstDate.setDate(firstDate.getDate() + 1);
+  }
+
+  const getDaysArray = function (start, end) {
+    for (
+      var gapDates = [], firstGapDate = new Date(start);
```
Comment (Sbonelo01) [inline frontend/src/components/regions/UserActions/utils.js]: Is there a specific reason for using `var`

**Review by Sbonelo01 — CHANGES_REQUESTED** [review]
Well done Themba! This seems to solve the problem - no dates seem to be getting skipped

- The `updateBurnDownSnapShots` function could be broken down further.

**Review by KhalidUmuzi — APPROVED** [review]

File: frontend/src/components/regions/UserActions/utils.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+export function updateBurnDownSnapshots({ burnDownSnapshots }) {
+  const datesArray = [];
+  burnDownSnapshots.forEach((burnDownSnapshot) => {
+    datesArray.push(burnDownSnapshot.timestamp);
+  });
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/utils.js:4]: This is good. I just want to guide you to writing code more efficiently. It'll save you time in the long run

Try using a `map`  to get the datesArray. Where else can map be useful?

File: frontend/src/components/regions/UserActions/utils.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+  const uniqueDatesArr = Array.from(new Set(datesArray)).sort(
+    (a, b) => new Date(a) - new Date(b)
+  );
+
+  const firstDate = new Date(new Date(uniqueDatesArr[0]).getTime());
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/utils.js]: I wonder if this code will error with an empty snapshot array. If the api call that fetches the data hasn't finished then the array might be empty.

Can you add another test that just makes sure nothing breaks when the array is empty?

File: frontend/src/components/regions/UserActions/UserBurndownStats.jsx [hunk 5]
```diff
@@ -49,24 +58,32 @@ export default ({ burnDownSnapshots }) => {
             name="Total Cards"
             stroke={orange[400]}
             activeDot={{ r: 8 }}
+            strokeWidth={1}
+            dot={false}
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/UserBurndownStats.jsx:62]: can you please include a screenshot in the pr description? I'm not sure what this will look like.

File: frontend/src/components/regions/UserActions/utils.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
+export function updateBurnDownSnapshots({ burnDownSnapshots }) {
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserActions/utils.js]: I wonder if there is a better name for this function. Always remember that good code makes its intentions clear.

Is this meant to fill in the gaps? Maybe `fillInSnapshotDateGaps` or something like that? 

And then if you are struggling to write self-documenting code, comments can be very helpful

**Review by sheenarbw — CHANGES_REQUESTED** [review]
This is good. There are a few little things worth looking at, mostly I thing there are a few lessons worth learning

File: frontend/src/components/regions/UserActions/utils.js (hunk 3 above)
Comment (KhalidUmuzi) [reply frontend/src/components/regions/UserActions/utils.js:4]: Have you tried to implement sheena's suggestion here? its not in the current changes if so

File: frontend/src/components/regions/UserActions/utils.js (hunk 4 above)
Comment (KhalidUmuzi) [reply frontend/src/components/regions/UserActions/utils.js]: Has this been addressed and checked? I see no new tests or changes to ensure this doesnt error out.

**Review by KhalidUmuzi — CHANGES_REQUESTED** [review]

**Comment by TheMaja** [comment]
@KhalidUmuzi  tried using your method by using burnDownSnapshots by removing the datesArray but it breaks everything apparently the datesArray is very important because it keeps the original burnDownSnapshots un-modified and thats very usefull down on the function.

**Review by KhalidUmuzi — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]
